src/cookbooks/events_best_practices.md

```diff
@@ -0,0 +1,101 @@
+
+One challenge with this data is it can be difficult to identify all the records from a particular event. Unlike Scalars and Histograms, which keep data in individual locations (like `scalar_parent_browser_engagement_total_uri_count` for [total_uri_count](https://dxr.mozilla.org/mozilla-central/source/toolkit/components/telemetry/Scalars.yaml#99)), all event records are stored together, regardless of which event generated them. The records themselves don't have a field identifying which event produced it[1].
+
+Take, for example, the ["manage"](https://dxr.mozilla.org/mozilla-central/source/toolkit/components/telemetry/Events.yaml#105)
+ event in the "addonsManager" category. 
+
+```
+addonsManager: # category
+  manage: # event name
+    description: >
+      ...
+    objects: ["extension", "theme", "locale", "dictionary", "other"] # object values
+    methods: ["disable", "enable", "sideload_prompt", "uninstall"] # method values
+    extra_keys: # extra values
+      ...
+    notification_emails: ...
+    expiry_version: ...
+    record_in_processes: ...
+    bug_numbers: ...
+    release_channel_collection: ...
+```
+This event will produce records that look like: 
+
+|timestamp| category | method  | object  | value | extra | 
+|-| -------- |---------| --------|-------|-------|
+|...|addonsManager|install|extension|        |...|
+|...|addonsManager|update|locale|        |...|
+|...|addonsManager|sideload_prompt|other|       |...|
+
+But none of these records will indicate that it was produced by the "manage" event. To find all records produced by "manage", one would have to query all records where 
+```
+category = ...
+AND method in [...,]
+AND object in [...,]
+```
+which is not ideal. 
+
+Furthermore, if one encounters this data without knowledge of how the "manage" event works, they need to look up the event based on the category, method, and object values in order to find the event, and then requery the data to find all the related events. It's not immediately clear from the data if this record: 
+
+|timestamp| category | method  | object  | value | extra | 
+|-| -------- |---------| --------|-------|-------|
+|...|addonsManager|update|locale|        |...|
+
+and and this record: 
+
+|timestamp| category | method  | object  | value | extra | 
+|-| -------- |---------| --------|-------|-------|
+|...|addonsManager|install|extension|        |...|
+
+are related or not. 
+
+Another factor that can add to confusion is the fact that other events can share similar values for methods or objects (or even the combination of method and object). For example:
+
+|timestamp| category | method  | object  | value | extra | 
+|-| -------- |---------| --------|-------|-------|
+|...|normandy|update|preference_rollout|        |...|
+
+which can further confuser users. 
+
+[1]: If a user defines an event in Events.yaml without specifying a list of acceptable methods, the method will default to the name of the event for records created by that event. 
```

Comment:
You may need to preface this footnote with "events do have `name` fields, but they aren't included in the event records and thus are not present in the resulting dataset. Also, If a user defines..."